### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by JulietShaban**

### CODE REVIEW

File: client/src/Routes/Routes.tsx [hunk 1]
```diff
@@ -31,6 +31,14 @@ const Routes: React.FC = () => {
                             </PublicRoute>
                         }
                     />
+                    <Route
+                        path={FORGOT_PATH}
+                        element={
+                            <PublicRoute>
+                                <ForgotPassword name="u" />
```
Comment (mykhalenych) [inline client/src/Routes/Routes.tsx]: what is name="u"?

File: client/src/Routes/constants.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export const AUTH_PATH = '/auth';
 export const LOGIN_PATH = `${AUTH_PATH}/login`;
+export const FORGOT_PATH = `${AUTH_PATH}/forgotpass`;
```
Comment (mykhalenych) [inline client/src/Routes/constants.ts]: bad naming 
```suggestion
export const FORGOT_PATH = `${AUTH_PATH}/forgot-password`;
```

File: client/src/layouts/AuthLayout/AuthLayout.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {SIGN_IN_PATH, LOGIN_PATH} from '../../Routes/constants';
 const AuthLayout = () => {
     return (
         <>
-            <ul>
+            {/* <ul>
```
Comment (mykhalenych) [inline client/src/layouts/AuthLayout/AuthLayout.tsx]: revert changes

File: client/src/layouts/MainLayout/MainLayout.tsx [hunk 4]
```diff
@@ -1,17 +1,25 @@
 import React from 'react';
 import {Outlet} from 'react-router-dom';
 
+import {IconButton} from '@mui/material';
```
Comment (mykhalenych) [inline client/src/layouts/MainLayout/MainLayout.tsx:3]: keep imports order, see readme.md

File: client/src/layouts/MainLayout/styles.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import {styled} from '@mui/material/styles';
 
+import {Link} from 'react-router-dom';
```
Comment (mykhalenych) [inline client/src/layouts/MainLayout/styles.ts:1]: the same

File: client/src/pages/Auth/ForgotPassword/ForgotPassword.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {useForm, FormProvider} from 'react-hook-form';
+import {Grid} from '@mui/material';
+
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/ForgotPassword.tsx]: the same

File: client/src/pages/Auth/Login/Login.tsx [hunk 7]
```diff
@@ -1,42 +1,62 @@
...
-                            Login
-                        </Button>
-                    </Grid>
+        <ContainerDiv>
+            <>
+                <LoginImg src={signInImage} alt="mountains" />
+                <Grid container flexDirection="column">
+                    <FormProvider {...methods}>
+                        <form onSubmit={handleSubmit(onSubmit)} noValidate>
+                            <Grid container justifyContent="center" height="100%">
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/Login.tsx:39]: delete height 100%

File: client/src/pages/Auth/Login/Login.tsx [hunk 8]
```diff
@@ -1,42 +1,62 @@
...
+                                    name="password"
+                                    fullWidth
+                                />
+                                <ForgotPassLink to={FORGOT_PATH}>Forgot password?</ForgotPassLink>
+                            </Grid>
+                            <Grid container justifyContent="center">
+                                <Button variant="outlined" type="submit" color="primary">
+                                    Login
+                                </Button>
+                                <SignLink to={'/'}> Don’t have an account? Sign up</SignLink>
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/Login.tsx]: to={SIGN_IN_PATH}

File: client/src/pages/Auth/Login/styles.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    maxWidth: 540,
+    maxHeight: 720,
+    left: '30%',
+    '& .MuiFormControl-root': {
+        margin: 0,
+    },
+    '& .MuiFormHelperText-root': {
+        margin: 0,
+    },
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:17]:
```suggestion
}));

```

File: client/src/pages/Auth/Login/styles.ts [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+export const LoginImg = styled('img')(({theme}) => ({
+    borderRadius: '20px 0 0 20px',
+    width: 220,
+    height: 720,
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:23]:
```suggestion
}));

```

File: client/src/pages/Auth/Login/styles.ts [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+export const Header = styled('h2')(({theme}) => ({
+    color: '#1C1B1F',
+    fontFamily: 'Georgia regular',
+    fontSize: 45,
+    fontWeight: 400,
+    height: 52,
+    marginTop: 115,
+    marginBottom: 0,
+    paddingBottom: 0,
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:34]:
```suggestion
}));

```

File: client/src/pages/Auth/Login/styles.ts [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+    color: '#1C1B1F',
+    boxSizing: 'border-box',
+    fontFamily: 'Roboto',
+    fontSize: 12,
+    height: '16px',
+    alignItems: 'center',
+    width: '100%',
+    textDecoration: 'none',
+    marginLeft: 40,
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:43]:
```suggestion
}));

```

File: client/src/pages/Auth/Login/styles.ts [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+export const SignLink = styled(Link)(({theme}) => ({
+    color: theme.palette.text.primary,
+    fontFamily: 'Roboto',
+    fontSize: 16,
+    display: 'inline-block',
+    width: 240,
+    marginTop: 56,
+    textDecoration: 'none',
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:51]:
```suggestion
}));

```

File: client/src/pages/Auth/Login/styles.ts [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+        margin: 0,
+    },
+}));
+export const LoginImg = styled('img')(({theme}) => ({
+    borderRadius: '20px 0 0 20px',
+    width: 220,
+    height: 720,
+}));
+export const Header = styled('h2')(({theme}) => ({
+    color: '#1C1B1F',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: use colors from pallete

File: client/src/pages/Auth/Login/styles.ts [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+    fontFamily: 'Georgia regular',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:27]: don't change fonts

File: client/src/pages/Auth/Login/styles.ts [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+    fontSize: 45,
+    fontWeight: 400,
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts:30]: it is by default

File: client/src/pages/Auth/Login/styles.ts [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+    height: 52,
+    marginTop: 115,
+    marginBottom: 0,
+    paddingBottom: 0,
+}));
+export const ForgotPassLink = styled(Link)(({theme}) => ({
+    color: '#1C1B1F',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: the same

File: client/src/pages/Auth/Login/styles.ts [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+    boxSizing: 'border-box',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: it must be in `*` selector for all styles

File: client/src/pages/Auth/Login/styles.ts [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+    fontFamily: 'Roboto',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: delete it

File: client/src/pages/Auth/Login/styles.ts [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+    fontSize: 12,
+    height: '16px',
+    alignItems: 'center',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: alignItems works only with display flex

File: client/src/pages/Auth/Login/styles.ts [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+    width: '100%',
+    textDecoration: 'none',
+    marginLeft: 40,
+}));
+export const SignLink = styled(Link)(({theme}) => ({
+    color: theme.palette.text.primary,
+    fontFamily: 'Roboto',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: the same

File: client/src/pages/Auth/Login/styles.ts [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+    fontSize: 16,
+    display: 'inline-block',
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: looks strange

File: client/src/pages/Auth/Login/styles.ts [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+    width: 240,
+    marginTop: 56,
+    textDecoration: 'none',
+}));
+export const CloseButton = styled(IconButton)(({theme}) => ({
+    marginLeft: 253,
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]: why 253? is isn't responsive, use display flex for parent and align content by justifyContent and alignItems

File: client/src/pages/Auth/SignIn/SignIn.tsx [hunk 24]
```diff
@@ -23,27 +24,35 @@ const SignIn = () => {
...
+                                fullWidth
+                            />
+                        </Grid>
 
-                    <Grid item xs={6}>
-                        <Button type="submit" variant="contained" color="primary">
-                            Sign In
-                        </Button>
+                        <Grid item xs={6}>
+                            <Button type="submit" variant="outlined" color="primary">
```
Comment (mykhalenych) [inline client/src/pages/Auth/SignIn/SignIn.tsx]: variant="outlined" keeps by default props you don't need to pass it

File: client/src/pages/Auth/Login/styles.ts [hunk 25]
```diff
@@ -0,0 +1,58 @@
+import {Link} from 'react-router-dom';
+
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/styles.ts]:
```suggestion
```

File: client/src/pages/Auth/ForgotPassword/ForgotPassword.tsx [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+const ForgotPassword: React.FC<ForgotPassProps> = () => {
+    const methods = useForm<ForgotPassProps>({
+        resolver: yupResolver(validation),
+    });
+    const {handleSubmit, control} = methods;
+
+    return (
+        <ContainerDiv>
+            <>
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/ForgotPassword.tsx:22]: why do you need fragment here?

File: client/src/pages/Auth/ForgotPassword/styles.ts [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+    maxWidth: 540,
+    maxHeight: 720,
+    left: '30%',
+    '& .MuiFormControl-root': {
+        margin: 0,
+    },
+    '& .MuiFormHelperText-root': {
+        margin: 0,
+    },
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/styles.ts:16]:
```suggestion
}));

```

File: client/src/pages/Auth/ForgotPassword/styles.ts [hunk 28]
```diff
@@ -0,0 +1,31 @@
...
+export const LoginImg = styled('img')(({theme}) => ({
+    borderRadius: '20px 0 0 20px',
+    width: 220,
+    height: 720,
+}));
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/styles.ts:22]:
```suggestion
}));

```

File: client/src/pages/Auth/ForgotPassword/styles.ts [hunk 29]
```diff
@@ -0,0 +1,31 @@
+import {styled} from '@mui/material/styles';
+
+export const ContainerDiv = styled('div')(({theme}) => ({
+    borderRadius: 20,
+    backgroundColor: '#F6F6F7',
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/styles.ts]: change. to pallet color

File: client/src/pages/Auth/ForgotPassword/styles.ts [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+        margin: 0,
+    },
+}));
+export const LoginImg = styled('img')(({theme}) => ({
+    borderRadius: '20px 0 0 20px',
+    width: 220,
+    height: 720,
+}));
+export const Header = styled('h2')(({theme}) => ({
+    color: '#1C1B1F',
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/styles.ts]: you can style h2 once in theme and don't duplicate code

File: client/src/pages/Auth/ForgotPassword/types.ts [hunk 31]
```diff
@@ -0,0 +1,3 @@
+export interface ForgotPassProps {
+    name: string;
```
Comment (mykhalenych) [inline client/src/pages/Auth/ForgotPassword/types.ts]: why name?
I think here must be email

File: client/src/pages/Auth/Login/Login.tsx [hunk 32]
```diff
@@ -1,42 +1,62 @@
...
 const Login = () => {
+    const dispatch = useAppDispatch();
     const methods = useForm<ILoginProps>({
         resolver: yupResolver(validation),
         defaultValues,
     });
     const {handleSubmit, control} = methods;
 
+    const onSubmit = (data: ILoginProps) => {
+        dispatch(logInUser(data));
```
Comment (mykhalenych) [inline client/src/pages/Auth/Login/Login.tsx]:
```suggestion
        dispatch(logInUser(data)).then((res) => if success redirect to main layout)
```

File: server/src/controllers/authController.js [hunk 33]
```diff
@@ -19,7 +19,7 @@ const register = async (req, res) => {
     ];
 
     if (errors.length) {
-        throw ApiError.BadRequest(errors.filter((item) => item));
+    throw ApiError.BadRequest(errors.filter((item) => item));
```
Comment (mykhalenych) [inline server/src/controllers/authController.js]: revert changes

**Review by mykhalenych — CHANGES_REQUESTED** [review]
Keep up the good work! Check comments

File: client/src/Routes/Routes.tsx (hunk 1 above)
Comment (JulietShaban) [reply client/src/Routes/Routes.tsx]: I created a random prop type(to be changed) for this component

File: client/src/Routes/constants.ts (hunk 2 above)
Comment (JulietShaban) [reply client/src/Routes/constants.ts]: on it

File: client/src/layouts/AuthLayout/AuthLayout.tsx (hunk 3 above)
Comment (JulietShaban) [reply client/src/layouts/AuthLayout/AuthLayout.tsx]: ok

File: client/src/layouts/MainLayout/MainLayout.tsx (hunk 4 above)
Comment (JulietShaban) [reply client/src/layouts/MainLayout/MainLayout.tsx:4]: ok

File: client/src/layouts/MainLayout/styles.ts (hunk 5 above)
Comment (JulietShaban) [reply client/src/layouts/MainLayout/styles.ts:3]: done

**Review by mykhalenych — APPROVED** [review]

**Pull request merged**